admin/includes/classes/AdminRequestSanitizer.php

```diff
@@ -588,19 +704,33 @@ private function filterStrictSanitizeValues()
      */
     private function traverseStrictSanitize(&$item, $ignore, $inner, $type)
     {
+        // Establish a local base array name for further processing.
+        $currentArrayName = $this->arrayName;
         foreach ($item as $k => $v) {
-            if ($inner || (!$inner && !in_array($k, $ignore))) {
+            // Append the key of this item to the arrayname that called the sanitizer.
+            $this->arrayName = $this->setCurrentArrayName($k, $currentArrayName);
+            if ($inner || (!$inner && !in_array($this->arrayName, $ignore))) {
                 if (is_array($v)) {
                     $item[$k] = $this->traverseStrictSanitize($v, $ignore, true, $type);
                 } else {
-                    $this->debugMessages[] = 'PROCESSING STRICT_SANITIZE_VALUES == ' . $k;
-                    $item[$k] = htmlspecialchars($item[$k]);
+                    if (!in_array($this->arrayName, $ignore)) {
+                        $this->debugMessages[] = 'PROCESSING STRICT_SANITIZE_VALUES == ' . $this->arrayName;
+                        $item[$k] = htmlspecialchars($item[$k]);
+                        if ($inner) {
+                            if ($type == 'post') {
+                                if (!in_array($this->arrayName, $ignore)) {
+                                    $this->postKeysAlreadySanitized[] = $this->arrayName;
+                                }
+                            }
+                        }
+                    }
                 }
             }
             if (!$inner) {
                 if ($type == 'post') {
-                    if (!in_array($k, $this->postKeysAlreadySanitized)) {
-                        $this->postKeysAlreadySanitized[] = $k;
+                    if (!in_array($this->arrayName, $this->postKeysAlreadySanitized)) {
+                        $this->postKeysAlreadySanitized[] = $this->arrayName;
```

Comment:
Why in filter's such as this is `setPostKeyAlreadySanitized($this->arrayName);` not used instead of manipulating the array directly? I some what get why `addKeyAlreadySanitized` wasn't selected to be used (if the key were already present then it wouldn't be added an additional time and it seems like it is desired to allow it to be added multiple times in the place(s) incorporated).